docs/development/TESTING.md

```
@@ -0,0 +1,526 @@
+# Run specific test
+pytest test/test_gui_manager.py::TestGUIManager::test_session_data
+```
+
+### Test Structure
+
+```
+ovos_gui/
+├── __init__.py
+├── manager.py        # Main GUIManager class
+├── session.py        # Session data handling
+├── bus_handlers.py   # MessageBus event handlers
```

markdownlint-cli2 (0.21.0)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)